### PR TITLE
Data processing wip

### PR DESCRIPTION
**Pull request by CuiHen**
Added dataset.py, a class to create a tensorflow dataset object to quickly iterate through the images.
At first, it loads every image into cache and allocates a label.
Afterwards, the images are preprocessed (from RGB int-range [0, 255] to float-range [-1, 1]), randomly shuffled and batched.

### CODE REVIEW

**Comment by CuiHen** [comment]
now with improved prefetch and no artificial flavourings

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,159 @@
+.idea/
+
```
Comment (dpaetzel) [inline .gitignore:4]: I prefer to add a line of the form

```
## From https://github.com/github/gitignore/blob/main/Python.gitignore
```

here so this is attributed to the gitignore repository correctly.

File: README.md [hunk 2]
```diff
@@ -29,3 +29,20 @@ long as we can clearly make them out, we say “they're there”.
 3. Train MobileNet.
 4. See how well MobileNet does.
 5. Try deploying MobileNet to David's Raspberry Pi 3.
+
+### TODO Mobilenet:
```
Comment (dpaetzel) [inline README.md]: Please leave *two* lines blank before and after headings in markdown and don't end headings in colons.

File: mobile_net/utils/dataset.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+# !/usr/bin/env python
+
+"""
+Loads all images into cache and create a tf.Dataset Object
```
Comment (dpaetzel) [inline mobile_net/utils/dataset.py]: Shouldn't this be the docstring of the `Dataset` class (or rather its `_get_dataset` method)?

File: mobile_net/utils/dataset.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# !/usr/bin/env python
```
Comment (dpaetzel) [inline mobile_net/utils/dataset.py]: If we want to do the whole “divide into multiple modules” thing (instead of just writing a few more scripts that we put into the root folder of the repository), *please use the src/module_name/ layout*.

But if you ask me, we should just write a few scripts in the repo root folder for now (we already have `sample_from_video_file.py`, why not just add the next stage(s) beside that?) and only after that think about whether we want to make this a library (which is probably not the case because it's rather restricted to this exact data set/learning task).

File: mobile_net/utils/dataset.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+Loads all images into cache and create a tf.Dataset Object
+"""
+import numpy as np
+import os
+import tensorflow as tf
+import cv2
+import tqdm
+
+
+class Dataset:
```
Comment (dpaetzel) [inline mobile_net/utils/dataset.py]: … based on my above-written doubts that we make this into a library in the end we may not need a class but instead just write the preprocessing pipeline at the start of the script that sets up, trains and exports the model?

File: mobile_net/utils/dataset.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    def __init__(self):
+        # TODO Henning: better pathing
+        self.img_folders = [r"../../both", r"../../no_falcons", r"../../tina", r"../../tom"]
```
Comment (dpaetzel) [inline mobile_net/utils/dataset.py]: This should probably be `r"both"` etc. for now since the script will be run from the root of the repository (i.e. the cwd will be the root of the repository)?

If this is turned into a script (see other comments) I will probably add `click`-based argument handling and make it possible to provide a path to the data.

**Review by dpaetzel — CHANGES_REQUESTED** [review]
tl;dr Let's turn this into a script. Other than that: Good work, thank you! :slightly_smiling_face:

**Review by dpaetzel — APPROVED** [review]
